src/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Ideally file renames are done in a separate commit without any other changes, so they show up as a rename and not delete\add.

Reply:
It wasn't a rename. In the first commit I added new classes. After that, in the second commit, all methods were moved out from the materializers to the accessors.

Reply:
Sure but a rename would have kept the history for `CreateConstructor` etc.

Reply:
In that case commits should be squashed since there are no additions in the second. Should I do that?

Reply:
No it was forward-looking, and mostly around true file renames. I'm guilty of this myself since I don't always know ahead of time how refactoring will end up, but I try to make reviewing easier for others and keep history when it makes sense.